src/service/url-replacements-impl.js

```diff
@@ -575,26 +580,30 @@ export class GlobalVariableSource extends VariableSource {
    */
   addReplaceParamsIfMissing_(orig) {
     const {replaceParams} =
-    /** @type {!Object} */ (Services.documentInfoForDoc(this.ampdoc));
+    /** @type {!Object} */ (this.getDocInfo_());
     if (!replaceParams) {
       return orig;
     }
     return addMissingParamsToUrl(removeAmpJsParamsFromUrl(orig), replaceParams);
   }
 
   /**
-   * Resolves the value via one of document info's urls.
-   * @param {string} field A field on the docInfo
-   * @param {string=} opt_urlProp A subproperty of the field
-   * @return {T}
-   * @template T
+   * Return the document info for the current ampdoc.
+   * @return {./document-info-impl.DocumentInfoDef}
    */
-  getDocInfoUrl_(field, opt_urlProp) {
-    return () => {
-      const docInfo = Services.documentInfoForDoc(this.ampdoc);
-      const value = docInfo[field];
-      return opt_urlProp ? parseUrlDeprecated(value)[opt_urlProp] : value;
-    };
+  getDocInfo_() {
+    return Services.documentInfoForDoc(this.ampdoc);
+  }
+
+  /**
+   * Return the parsed url for doc.
+   * @param {string} url
+   * @param {boolean=} opt_nocache
+   * @return {!Location}
+   */
+  parseUrl_(url, opt_nocache) {
+    return Services.urlForDoc(this.ampdoc.getHeadNode())
```

Comment:
Are we sure this will resolve relative URLs in the right context for FIE? `ampdoc.getHeadNode()` will be use the parent window context, whereas `parseUrlDeprecated` uses `self.document`.

Reply:
Interesting, is there a preferred way to get the service now? I saw your change that you now need an element, and we don't have easy access to one here (that I know of).

Reply:
Actually does it even matter as we are passing the url to the `parse()` call, not relying on the service for the url.

Reply:
Correct, passing the element is the best way. Might require some refactoring if you want to make this change.

Reply:
Oh I see this is for the other cases, not just the new stuff.

Reply:
changed back to `parseUrlDeprecated`.